secretsmanager/src/main/scala/uk/gov/nationalarchives/DASecretsManagerClient.scala

```diff
@@ -0,0 +1,174 @@
+package uk.gov.nationalarchives
+
+import cats.effect.Async
+import cats.implicits._
+import io.circe.Printer.noSpaces
+import io.circe._
+import io.circe.parser.decode
+import io.circe.syntax._
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerAsyncClient
+import software.amazon.awssdk.services.secretsmanager.model._
+import uk.gov.nationalarchives.DASecretsManagerClient._
+
+import java.util.concurrent.CompletableFuture
+
+/** A secrets manager client. It is written generically so can be used for any effect which has an Async instance.
+  * Requires an implicit instance of cats Async which is used to convert CompletableFuture to F
+  * @param secretsManagerAsyncClient
+  *   An AWS secrets manager async client
+  * @param secretId
+  *   The id of the secret. You will need multiple client instances for multiple secrets
+  * @tparam F
+  *   The type of the effect
+  */
+class DASecretsManagerClient[F[_]: Async](secretsManagerAsyncClient: SecretsManagerAsyncClient, secretId: String) {
+
+  implicit class CompletableFutureUtils[T](completableFuture: CompletableFuture[T]) {
+    def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
+  }
+
+  /** Generates a random password
+    * @param passwordLength
+    *   The length of the password. Defaults to 15
+    * @param excludeCharacters
+    *   Characters to exclude from the password, defaults to ' and ""
```

Comment:
It's hard to see in the string but it actually defaults to `'`, `"` and `\` as for some weird reason, AWS include backslashes when generating passwords

Reply:
I've changed it to `defaults to ' and "" and \` I tried `defaults to ',  "" and \` but that looked weird with the comma in there.